### PR TITLE
Migrate Status component to the SDK

### PR DESCRIPTION
**Pull request by fhlavac**
Migrate `Status` component to the SDK as a part of ListPage migration effort
https://issues.redhat.com/browse/HAC-311

@invincibleJai @andrewballantyne

### CODE REVIEW

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/statuses.tsx
Comment (fhlavac) [reply frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/statuses.tsx]: @invincibleJai 
Made `title` property required and moved the translation back to console using a separate wrapper components.

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/Status.tsx
Comment (invincibleJai) [inline frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/Status.tsx:11]: we should add `@patternfly/react-icons` to shared-modules and `shared-modules-init`

https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/shared-modules.ts#L4

https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/shared-modules-init.ts#L6

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/StatusIconAndText.tsx
Comment (invincibleJai) [inline frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/StatusIconAndText.tsx:20]: why not continue with const?

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/statuses.tsx
Comment (invincibleJai) [inline frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/statuses.tsx:8]: we can let props go in the flow as they both have `title` unless we are not updating it, i don't think we need to pass title explictly 

```suggestion
  <GenericStatus {...props} Icon={RedExclamationCircleIcon}  />
```

File: frontend/packages/console-dynamic-plugin-sdk/src/api/internal-api-kubevirt.ts
Comment (invincibleJai) [inline frontend/packages/console-dynamic-plugin-sdk/src/api/internal-api-kubevirt.ts:359]: as you migrated `GenericStatus` to SDK itself, so do we still need to lazy load it?

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/StatusIconAndText.tsx
Comment (fhlavac) [reply frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/StatusIconAndText.tsx:20]: @invincibleJai  We definitely can, just not sure why we need also a const for something like a dash. I thought it is more some kind of a remnant from the past.

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/statuses.tsx
Comment (fhlavac) [reply frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/statuses.tsx:8]: @invincibleJai it then does not comply with the Icon proptypes:
```
type GenericStatusProps = StatusComponentProps & {
  Icon: React.ComponentType<{ title?: string }>;
  title: string;
  popoverTitle?: string;
  noTooltip?: boolean;
};
```
`Type '{ Icon: FC<ColoredIconProps>; title?: string; iconOnly?: boolean; noTooltip?: boolean; className?: string; popoverTitle?: string; children?: ReactNode; }' is not assignable to type '{ Icon: ComponentType<{ title?: string; }>; ...`

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/Status.tsx
Comment (fhlavac) [reply frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/Status.tsx:11]: @invincibleJai trying to find an answer but still not sure, do we need to list it if it's a dependency of `@patternfly/react-core` ?
now, the build fails with `Failed to parse package dependencies: @patternfly/react-icons
`

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/StatusIconAndText.tsx
Comment (fhlavac) [reply frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/StatusIconAndText.tsx:20]: @andrewballantyne gotha, sounds good, thank you :slightly_smiling_face:

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/Status.tsx
Comment (invincibleJai) [reply frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/Status.tsx:11]: ah it's part of `@patternfly/react-core` then ignore my comment :)

**Comment by invincibleJai** [comment]
@fhlavac  can you squash the commits

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/Status.tsx
Comment (invincibleJai) [inline frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/Status.tsx:107]: nit: can have type above component

**Comment by fhlavac** [comment]
@andrewballantyne what do you think, can we add qe-approve ourselves? Thank you

File: frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/Status.tsx
Comment (invincibleJai) [inline frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/Status.tsx:22]: I believe we are not exposing these components (status/StatusIconAndText etc) from SDK via `api` as of now as we would only export `ListPageHeader` so it's fine but it would be nice to add some jsdocs for these if we plan to expose

https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts

**Comment by invincibleJai** [comment]
/approve

overall looks good, I don't see any breaking change. The components moved as part of this PR are needed to migrate `ListPageHeader`.

cc @andrewballantyne

**Comment by andrewballantyne** [comment]
> @andrewballantyne what do you think, can we add qe-approve ourselves? Thank you

I'd like not to do anything ourselves this close to 4.10 (next Friday). We can wait until the 31st and merge with our approval, or we get QE on it and do a test of the console. Breaking anything now would not work out well for GA.

**Review by andrewballantyne — APPROVED** [review]
/lgtm

Looking good. It is a bit of a refactor but looks pretty safe.

Adding doc labels as it's an internal refactor and we do not need anything on that front.
/label docs-approved
/label px-approved

**Comment by andrewballantyne** [comment]
/cc @yapei 

The testing strategy imo (@fhlavac / @invincibleJai feel free to add/correct) would be to test the console in various spots where objects show status. I've tested Builds/Pipelines/Pods/Project Details Pages (and ODC Project Dashboard) without seeing any issue. We should probably deploy the plugin and test that out as well, but I'm not sure there is any status usages in the test pages -- so it's probably a sanity check more than anything else.

**Comment by andrewballantyne** [comment]
/hold

Waiting until Jan 31st to merge at the earliest. Reducing churn on 4.10 effort.

**Review by andrewballantyne — APPROVED** [review]
/lgtm

Migration of component, don't need direct QE testing. I have tested the various spots we render it and it looks un-impacted by this PR.
/label qe-approved

**Pull request merged**
